lib/ecto/changeset.ex

```diff
@@ -351,6 +352,15 @@ defmodule Ecto.Changeset do
   @type data :: map()
   @type types :: map()
 
+  @typedoc """
+  A possible value that you can pass to the `:empty_values` option.
+
+  See `empty_values/0` and the [*Empty values* section](#module-empty-values) in
+  the module documentation for more information.
+  """
+  @typedoc since: "3.11.0"
+  @type empty_value :: (term() -> boolean()) | term()
```

Comment:
`term()` already includes anonymous functions. 🤔

Reply:
Maybe we want to break term into strings and all other composite data types?

Reply:
@josevalim yes `term()` contains everything ofc, but I think this for documentation is a decent approach. Otherwise, we can also do `(term -> boolean) | String.t | term`, kind of the same. Or you meant something different?

Reply:
I was thinking of:

```suggestion
  @type empty_value :: (term() -> boolean()) | binary() | list() | map() | tuple()
```

Cause i can't think of anything else that would be reasonable as empty value?

Reply:
Why not atoms, or numbers (like `0`)?

Reply:
I can't see a reason where it would be sane to consider any of those as empty values.

Reply:
Right and actually, folks can use a function if they wanna do something weird, so let's go for that. 👍